File: backend/src/data/globalChatStickers.ts
export type GlobalChatSticker = {
  id: string;
  emoji: string;
};

/** Curated world-chat stickers. Same ids as the Flutter catalog. */
export const GLOBAL_CHAT_STICKERS: GlobalChatSticker[] = [
  { id: 'cash', emoji: '💵' },
  { id: 'gun', emoji: '🔫' },
  { id: 'skull', emoji: '💀' },
  { id: 'fire', emoji: '🔥' },
  { id: 'car', emoji: '🚗' },
  { id: 'plane', emoji: '✈️' },
  { id: 'crown', emoji: '👑' },
  { id: 'dice', emoji: '🎲' },
  { id: 'heart', emoji: '❤️' },
  { id: 'laugh', emoji: '😂' },
  { id: 'cool', emoji: '😎' },
  { id: 'angry', emoji: '😡' },
  { id: 'thumbs', emoji: '👍' },
  { id: 'clap', emoji: '👏' },
  { id: 'cop', emoji: '👮' },
  { id: 'lock', emoji: '🔒' },
  { id: 'bomb', emoji: '💣' },
  { id: 'diamond', emoji: '💎' },
  { id: 'cheers', emoji: '🥃' },
  { id: 'rose', emoji: '🌹' },
  { id: 'smoke', emoji: '🚬' },
  { id: 'night', emoji: '🌃' },
  { id: 'wolf', emoji: '🐺' },
  { id: 'eye', emoji: '👁️' },
];

const BY_ID = new Map(GLOBAL_CHAT_STICKERS.map((sticker) => [sticker.id, sticker]));

export function getGlobalChatSticker(id: string | null | undefined): GlobalChatSticker | null {
  if (!id) return null;
  return BY_ID.get(id) ?? null;
}

export function isGlobalChatStickerId(id: string | null | undefined): boolean {
  return Boolean(id && BY_ID.has(id));
}

File: client/lib/data/global_chat_stickers.dart
class GlobalChatSticker {
  final String id;
  final String emoji;

  const GlobalChatSticker({required this.id, required this.emoji});
}

const List<GlobalChatSticker> kGlobalChatStickers = [
  GlobalChatSticker(id: 'cash', emoji: '💵'),
  GlobalChatSticker(id: 'gun', emoji: '🔫'),
  GlobalChatSticker(id: 'skull', emoji: '💀'),
  GlobalChatSticker(id: 'fire', emoji: '🔥'),
  GlobalChatSticker(id: 'car', emoji: '🚗'),
  GlobalChatSticker(id: 'plane', emoji: '✈️'),
  GlobalChatSticker(id: 'crown', emoji: '👑'),
  GlobalChatSticker(id: 'dice', emoji: '🎲'),
  GlobalChatSticker(id: 'heart', emoji: '❤️'),
  GlobalChatSticker(id: 'laugh', emoji: '😂'),
  GlobalChatSticker(id: 'cool', emoji: '😎'),
  GlobalChatSticker(id: 'angry', emoji: '😡'),
  GlobalChatSticker(id: 'thumbs', emoji: '👍'),
  GlobalChatSticker(id: 'clap', emoji: '👏'),
  GlobalChatSticker(id: 'cop', emoji: '👮'),
  GlobalChatSticker(id: 'lock', emoji: '🔒'),
  GlobalChatSticker(id: 'bomb', emoji: '💣'),
  GlobalChatSticker(id: 'diamond', emoji: '💎'),
  GlobalChatSticker(id: 'cheers', emoji: '🥃'),
  GlobalChatSticker(id: 'rose', emoji: '🌹'),
  GlobalChatSticker(id: 'smoke', emoji: '🚬'),
  GlobalChatSticker(id: 'night', emoji: '🌃'),
  GlobalChatSticker(id: 'wolf', emoji: '🐺'),
  GlobalChatSticker(id: 'eye', emoji: '👁️'),
];

GlobalChatSticker? globalChatStickerById(String? id) {
  if (id == null) return null;
  for (final sticker in kGlobalChatStickers) {
    if (sticker.id == id) return sticker;
  }
  return null;
}
